spec/react.spec.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+
+const assert = require('assert');
+const sinon = require('sinon');
+const React = require('react');
+const ReactTestRenderer = require('react-test-renderer');
+
+const equal = require('../index');
+const tests = require('./tests');
+
+class ChildWithShouldComponentUpdate extends React.Component {
+  shouldComponentUpdate(nextProps) {
+    // this.props.children is a div with a circular reference to its owner, Container
+    return !equal(this.props, nextProps);
+  }
+  render() {
+    return null;
+  }
+}
+
+class Container extends React.Component {
+  render() {
+    return React.createElement(ChildWithShouldComponentUpdate, {
+      children: React.createElement('div')
+    });
+  }
+}
+
+describe('advanced', () => {
+  let warnStub = sinon.stub(console, 'warn');
```

Comment:
GLOBAL: Never _set_ something in a `describe`. Leave it open like:

```js
let warnStub;
```

Then set in `beforeEach`.

Reply:
GLOBAL: Technically optional, but **strongly** recommended. Use a `sandbox` instead. Sandboxes track fakes and globally restore saving you the pain of having to manually restore. So, here putting together all the comments:

```js
describe('advanced', () => {
  let sandbox;
  let warnStub;

  beforeEach(() => {
    sandbox = sinon.sandbox.create();
    warnStub = sandbox.stub(console, "warn");
  });

  afterEach(() => {
    sandbox.restore();
  });
```

You can move that sandbox creation to a higher level `beforeEach` or even a base-but-always-included global `beforeEach|afterEach` too if you need fakes across a test suite -- reusing the _same_ sandbox is a good practice and I can help fill in the logistical details for approaches for this if you need this in the future 😉

Reply:
sounds great!